lighthouse-core/gather/gatherers/seo/tap-targets.js

```diff
@@ -257,10 +230,18 @@ function gatherTapTargets() {
   /** @type {LH.Artifacts.TapTarget[]} */
   const targets = [];
 
+  // Capture element positions relative to the top of the page
+  window.scrollTo(0, 0);
```

Comment:
one issue is what if the page purposely scrolled down? Seems similar to the axe gathering, e.g. if we took a screenshot (for some reason) in a gatherer after this one, as a user I'd expect it to be the targeted final look of the page, not scrolled back to the top. Does this gatherer need to be last (or next to last) if it wants to change  scroll position?

(of course there are lots of issues with actually supporting this that we don't account for in e.g. offscreen-images)

Reply:
hm I wonder why it's not stacking our convos...

anyhoo :)

I disagree we need to be worried about always returning the scroll position to the exact same spot as before. I think it's a far more reasonable contract that every gatherer should scroll to where it needs to be and not assume anything about scroll position and we reset scroll position to 0,0 between every gatherer. It doesn't preclude anyone from taking screenshots of a particular scrolled area with the protocol unless I'm misunderstanding your objection?

Reply:
> It doesn't preclude anyone from taking screenshots of a particular scrolled area with the protocol unless I'm misunderstanding your objection?

FWIW, I meant if the page scrolled itself (e.g. it's an SPA with a router that scrolls to a particular part of the page based on whatever), but I'm not sure if that changes much about your point

Reply:
> it's an SPA with a router that scrolls to a particular part of the page based on whatever

Ah, gotcha. Well I think my point remains unchanged but replace "we reset scroll position to 0,0 between every gatherer" with "we reset scroll position to endOfPassX,endOfPassY between every gatherer" :)